ensembl_genes/Red.py

```diff
@@ -197,42 +196,41 @@ def write_output(self):
         repeats_file = self.parse_repeats(self.param('rpt'),repeat_consensus_id,analysis_id)
 
         # insert repeat features
-        repeat_feature_query = "LOAD DATA LOCAL INFILE '"+repeats_file+"' \
-                                INTO TABLE repeat_feature \
-                                FIELDS TERMINATED BY '\\t' \
-                                LINES TERMINATED BY '\\n' \
-                                (seq_region_id,seq_region_start,seq_region_end,\
-                                 repeat_start,repeat_end,repeat_consensus_id,analysis_id)"
+        repeat_feature_query = f'LOAD DATA LOCAL INFILE "{repeats_file}"' \
+                               +"INTO TABLE repeat_feature \
+                                 FIELDS TERMINATED BY '\\t' \
+                                 LINES TERMINATED BY '\\n' \
+                                 (seq_region_id,seq_region_start,seq_region_end, \
+                                  repeat_start,repeat_end,repeat_consensus_id,analysis_id)"
         connection.execute(repeat_feature_query)
 
         # delete temporary directory and its contents
         try:
             shutil.rmtree(self.param('gnm'))
         except OSError as e:
-            print("Error: %s : %s" % (self.param('gnm'),e.strerror))
+            print(f'Error: {self.param("gnm")} : {e.strerror}')
 
     def parse_repeats(self,rpt,repeat_consensus_id,analysis_id):
         """ It parses the Red's program output and it converts it into
             a tsv file which can be loaded into an Ensembl core repeat_feature table. """
-        rpt_files = os.listdir(rpt)
+        rpt_files = [str(path) for path in Path(rpt).iterdir()]
```

Comment:
Nice list comprehension! Maybe you can leave them as Path objects?

Reply:
I think I was trying to count the number of paths using Path but I could not find any way to do it simpler. I have had another look at this part of the code and I have removed the checks because they are not necessary any more since I delete the dir and create it again following the previous changes.